File: Java/Solution_151.java
public class Solution_151 {
    public String reverseWords(String s) {
        s = s.trim();
        String[] words = s.split("\\s+");

        StringBuilder reversed = new StringBuilder();
        
        for (int i = words.length - 1; i >= 0; i--) {
            reversed.append(words[i]);
            if (i != 0) {
                reversed.append(" ");
            }
        }
        return reversed.toString();
    }

    public static void main(String[] args) {
        Solution_151 solution = new Solution_151();

        System.out.println(solution.reverseWords("the sky is blue"));  // Output: "blue is sky the"
        System.out.println(solution.reverseWords("  hello world  "));  // Output: "world hello"
        System.out.println(solution.reverseWords("a good   example")); // Output: "example good a"

    }
}

File: Python/Solution-151.py
class Solution_151(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: int
        """
        s = s.strip()
        
        words = s.split()
        
        words.reverse()
        
        return ' '.join(words)
    
if __name__ == "__main__":
    solution = Solution_151()
    
    test_cases = [
        "the sky is blue",                  
        "  hello world  ",    
        "a good   example"
    ]
    
    for s in test_cases:
        print(f"Input: {s}")
        print(f"Output: {solution.reverseWords(s)}\n")